BedrockServer.cpp

```diff
@@ -2209,20 +2035,268 @@ void BedrockServer::_finishPeerCommand(unique_ptr<BedrockCommand>& command) {
+    // Initialize and get a unique thread ID.
+    SInitialize("socket" + to_string(_socketThreadNumber++));
+    SINFO("Socket thread starting");
+
+    // This outer loop just runs until the entire socket life cycle is done, meaning it deserializes a command,
+    // waits for it to get processed, deserializes another, etc, until the socket gets closed.
+    // This whole block is largely duplicated from `postPoll` and modified to work on a single non-blocking socket.
+    while (1) {
+        // We are going to call `poll` in a loop with only this one socket as a file descriptor.
+        // The reason for this is because it's possible that a client is connected to us, and not sending us any data.
+        // It may be waiting for it's own data before it can send us a request, or it may have just forgotten to
+        // disconnect. In the normal case, this is no big deal, we can wait inside `recv` until it either sends us some
+        // data or it disconnects. The exception is if we want to shut down. In that case, we need to know to close the
+        // socket at some point, so what we do is `poll` with a 1 second timeout, and if we ever hit the timeout and
+        // are in a `shutting down` state, then we finish up and exit. In any other case, we just wait in `poll` again
+        // until we get some data or a disconnection.
+        int pollResult = 0;
+        bool exitMainLoop = false;
+        struct pollfd pollStruct = { s->s, POLLIN, 0 };
+
+        // This loop just calls `poll` sequentially optionally exiting if we're shutting down and haven't received any
+        // data.
+        while (true) {
+            // Do the actual `poll` call.
+            pollResult = poll(&pollStruct, 1, 1'000);
+
+            // A result > 0 indicates a socket is ready to read (or has an error).
+            if (pollResult > 0) {
+                // We can now safely call `recv` without blocking. If it returns true, we've got some data, so we'll
+                // break out of this loop and see if we can do something with that data (like create a command).
+                if (!s->recv()) {
+                    // But if it returns false, this means our connection is dead and we should clean it up. We'll drop
+                    // to the bottom of the whole function here.
+                    exitMainLoop = true;
+                }
+                // We break out of the inner loop in either case.
+                break;
+            } else if (pollResult == 0) {
+                // Timeout. If we're shutting down, exit main loop, otherwise, try again.
+                if (_shutdownState.load() != RUNNING) {
+                    SINFO("Socket thread exiting because no data and shutting down.");
+                    exitMainLoop = true;
+                    break;
+                }
+            } else {
+                // This is an exceptional case, we'll just kill the socket if this happens and let the client
```

Comment:
NAB I'm not sure what makes the case exceptional, it's not wining any gold stars in my book.
```suggestion
                // This is an exception case, we'll just kill the socket if this happens and let the client
```